include/swift/Reflection/ReflectionContext.h

```diff
@@ -205,7 +205,36 @@ class ReflectionContext
       return false;
 
     auto Slide = ImageStart.getAddressData() - Command->vmaddr;
+    std::string Prefix = "__swift5";
+    uint64_t RangeStart = UINT64_MAX;
+    uint64_t RangeEnd = UINT64_MAX;
     auto SectionsBuf = reinterpret_cast<const char *>(Sections.get());
+    for (unsigned I = 0; I < NumSect; ++I) {
+      auto S = reinterpret_cast<typename T::Section *>(
+          SectionsBuf + (I * sizeof(typename T::Section)));
+      if (strncmp(S->sectname, Prefix.c_str(), strlen(Prefix.c_str())) != 0)
+        continue;
+      if (RangeStart == UINT64_MAX && RangeEnd == UINT64_MAX) {
+        RangeStart = S->addr + Slide;
+        RangeEnd = S->addr + S->size + Slide;
+        continue;
+      }
+      RangeStart = std::min(RangeStart, (uint64_t)S->addr + Slide);
+      RangeEnd = std::max(RangeEnd, (uint64_t)(S->addr + S->size + Slide));
+      // Keep the range rounded to 8 byte alignment on both ends so we don't
+      // introduce misaligned pointers mapping between local and remote
+      // address space.
+      RangeStart = RangeStart & ~7;
```

Comment:
LLVM's MathExtras.h has a couple of helper functions https://llvm.org/doxygen/MathExtras_8h_source.html that might make this easier read and the intention clearer.